### ISSUE TITLE
Deprecate dummy mode & make dummy-local work using a special job platform

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-flow#3482]
## Problem
Dummy and dummy-local modes modify suite config `[runtime][TASK]` section that will be moved to a platform in the globalconfig file in Cylc 8. We cannot easily or safely modify the global config object because if we reload it any changes will disappear.

Dummy mode is effectively a residual test battery for the batch sys handlers and should be deprecated because it is very easy to test all things remote with a simple hand-rolled test suite.

[Dummy-local mode will be retained - it is sufficient for testing everything except remote job host connectivity ... I guess we should rename it it to "dummy mode" though, once the current non-local dummy mode is gone.]

## Proposed changes.
1. Deprecate dummy mode.

**Comment by wxtim** [comment]
@oliver-sanders  said
> Just spotted a platforms issue. When you run with --mode=dummy it adds platform=dummy. If the suite contains stuff like [remote]host or [job]batch system this will result in mixed syntax which causes the upgrader to fail and raise a PlatformLookupError.

**Comment by wxtim** [comment]
@oliver-sanders  - Is this necessary - it looks to me like localhost will do.

> Create a hardcoded job platform in the globalconfig's SPEC['job platforms'] dictionary called dummy-local with suitable settings for dummy local running.

**Comment by oliver-sanders** [comment]
The goal of dummy-local mode is to reconfigure the task to run a `sleep` job locally. Doesn't matter massively how it's implemented so long as it ticks those boxes.

Note that a task has multiple scripting items (e.g. `pre-script`) and that environment variables can contain scripting too.

**Comment by wxtim** [comment]
> Note that a task has multiple scripting items (e.g. pre-script) and that environment variables can contain scripting too.

Do we have tests that these are handled?

**Comment by oliver-sanders** [comment]
> Do we have tests that pre-script &c are handled

`git grep dummy` yields a few results including `tests/flakyfunctional/modes/03-dummy-env.t`

**Comment by hjoliver** [comment]
> > Do we have tests that pre-script &c are handled
> 
> `git grep dummy` yields a few results including `tests/flakyfunctional/modes/03-dummy-env.t` which apparently tests script items.

```bash
# Test that user environment is disabled along with env-script in dummy mode.  
```

That could easily be extended to check the other script items like you suggest, @wxtim.

**Comment by wxtim** [comment]
Closed with PR #3841
